.circleci/config.yml

```diff
@@ -53,7 +53,14 @@ commands:
           at: /tmp
       - run:
           name: 'Maybe Gracefully Halt'
-          command: if ls /tmp/workspace/.CI_GRACEFULLY_HALT_* 1>/dev/null 2>&1; then echo "Gracefully halting this job."; circleci-agent step halt; fi
+          command: |
```

Comment:
This step has now become complex enough that I think it's time to move it into its own shell script and invoke it here via `curl`, similar to `'Compute Merge Commit'`. The main reason is that every edit to this file will force all in-flight PRs across the repo to rebase their PR branches, while edits to the shell script will not.

Reply:
I'm worried that using curl-and-execute would make this _harder_ to update. A PR that changes the bash file can't test itself because it'll still fetch the bash file from the main branch. How does that work?

Reply:
Alright, I see your point... Let's do this "properly" by splitting this into two PRs

I think the `mv workspace` command should stay its own hard-coded step (because it's extremely CircleCI specific and could be confusing to future readers if it's not explicitly there), but the rest of the graceful-halt can be moved to a curl'd file. Agree?

Reply:
I still say we should move all steps into a shell script and change this command to a single `curl` invocation. The rationale for this is in the screenshot below:

![image](https://user-images.githubusercontent.com/26553114/114212400-ca364900-992f-11eb-9356-5bf1752449bc.png)

All our CircleCI specific business logic lives in these shell files. It has two advantages:
- Makes `config.yml` a lot less compact and manageable
- Eliminates the need for wholesale rebasing of in-flight PRs when changes are restricted to the shell scripts
- Since these files all live in the `.circleci/` directory, it's already clear they shouldn't be used anywhere else.

WDYT? 😃

Reply:
Tested this PR by curling from my own branch and the [CI passed](https://app.circleci.com/pipelines/github/ampproject/amphtml/6477/workflows/acf8d6ef-45d5-4057-8095-cf2ea7096e19), I'll now push a commit with the corrected URL which will cause this PR to fail until we merge #33734

Reply:
Missed your comment above. PTAL at the config file, do you think it makes sense to add the `mv` command into `maybe_gracefully_halt.sh`? It seems like the wrong place and would be unexpected for the unaware reader

Reply:
Good question. How about combining the two steps, roughly like this?

```yml
  restore_workspace_and_maybe_halt:
    steps:
      - attach_workspace:
          at: /tmp
      - run:
          name: 'Restore Workspace and Maybe Halt'
          command: curl -sS https://raw.githubusercontent.com/ampproject/amphtml/main/.circleci/restore_workspace_and_maybe_halt.sh | bash
```

Reply:
I feel pretty strongly that the rename should be its own thing :/

Reply:
No worries, I'm happy to defer to you on this one. I think you've already done plenty to address the top-level concern of minimizing changes to the root `config.yml`.